ProjFS.Mac/PrjFSKext/KauthHandler.cpp

```diff
@@ -273,8 +471,8 @@ KEXT_STATIC int HandleVnodeOperation(
                 KAUTH_VNODE_SEARCH |
                 KAUTH_VNODE_READ_SECURITY |
                 KAUTH_VNODE_READ_ATTRIBUTES |
-                KAUTH_VNODE_READ_EXTATTRIBUTES |
-                KAUTH_VNODE_DELETE))
+                KAUTH_VNODE_READ_EXTATTRIBUTES) ||
+            (isDeleteAction && isRename))
         {
             // Recursively expand directory on delete to ensure child placeholders are created before rename operations
             if (isDeleteAction)
```

Comment:
I've just been struck by a thought: have we even been handling this case correctly? (mostly unrelated to this change) If we move a directory out of a repo, and it's recursively enumerated, won't it just end up full of empty files? I didn't think recursive enumeration hydrated files.

Reply:
> If we move a directory out of a repo, and it's recursively enumerated, won't it just end up full of empty files? I didn't think recursive enumeration hydrated files.

Yep this has been broken and currently you can only rename directories within a repo.  We had #515 open to fix it, but then we re-purposed #515 to say that we should block renames of projected folders.

Rather than fixing the scenario you've called out by implementing recursive hydration the plan is to set the minimum version to Mojave as soon as we can and then block renames of projected folders.

Reply:
Ahh, right, that was the reason for wanting to block them. I can integrate that blocking in my work on #182 for Mojave+ only.